src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/deciders/Publisher.java

```diff
@@ -18,33 +18,59 @@
 import com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions.Action;
 import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.core.NonLeafNode;
 import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.scheduler.FlowUnitOperationArgWrapper;
+
+import java.time.Instant;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
 public class Publisher extends NonLeafNode<EmptyFlowUnit> {
 
   private static final Logger LOG = LogManager.getLogger(Publisher.class);
+  private final long initTime;
 
   private Collator collator;
   private boolean isMuted = false;
+  private Map<String, Long> actionToExecutionTime;
 
   public Publisher(int evalIntervalSeconds, Collator collator) {
     super(0, evalIntervalSeconds);
     this.collator = collator;
+    this.actionToExecutionTime = new ConcurrentHashMap<>();
+    initTime = Instant.now().toEpochMilli();
+  }
+
+  /**
+   * Returns true if a given {@link Action}'s last execution time was > {@link Action#coolOffPeriodInMillis()} ago
+   *
+   * <p>If this Publisher has never executed the action, the last execution time is defined as the time that the publisher
+   * object was constructed.
+   *
+   * @param action The {@link Action} to test
+   * @return true if a given {@link Action}'s last execution time was > {@link Action#coolOffPeriodInMillis()} ago
+   */
+  public boolean isCooledOff(Action action) {
+    long lastExecution = actionToExecutionTime.getOrDefault(action.name(), initTime);
+    long now = Instant.now().toEpochMilli();
+    return now - lastExecution > action.coolOffPeriodInMillis();
   }
 
   @Override
   public EmptyFlowUnit operate() {
-    // TODO: Pass through implementation, need to add dampening, cool-off, action flip-flop
+    // TODO: Pass through implementation, need to add dampening, action flip-flop
     // avoidance, state persistence etc.
-
     Decision decision = collator.getFlowUnits().get(0);
     for (Action action : decision.getActions()) {
-      LOG.info("Executing action: [{}]", action.name());
-      action.execute();
+      if (isCooledOff(action)) { // Only execute actions which have passed their cool off period
+        LOG.info("Executing action: [{}]", action.name());
+        action.execute();
+        actionToExecutionTime.put(action.name(), Instant.now().toEpochMilli());
+      }
```

Comment:
Minor: Let's add a debug log around time remaining for cool-off to complete.

Reply:
Added to isCooledOff()